README
# supagrep

A simple sourcecode search tool for windows. It's something like 

 find . -name "*.c" | xargs grep printf
 
but for windows. This is not a command line tool.

CROSSCOMPILATION FOR WIN
=========================

I'm using opensuse leap 15.4
* install mingw64 cross toolchain
  zypper addrepo https://download.opensuse.org/repositories/windows:mingw:win64/openSUSE_Leap_15.3/windows:mingw:win64.repo
  zypper refresh
  zypper install mingw64-cross-gcc-c++ \
                 mingw64-cross-pkgconf \
                 mingw64-libjpeg-devel \
                 mingw64-libpng-devel \
                 mingw64-zlib-devel \
                 mingw64-libjpeg8 \
                 mingw64-libbz2-1 \
                 mingw64-libpng16-16 \
                 mingw64-gettext-tools

* cross compile the FOX library
  wget ftp://fox-toolkit.org/pub/fox-1.7.78.tar.gz
  tar xvfz fox-1.7.78.tar.gz
  cd fox-1.7.78
  ./configure --without-opengl \
    --host=x86_64-w64-mingw32 \
    --without-x \
    --without-xft \
    --prefix=/usr/x86_64-w64-mingw32/sys-root/mingw/ \
    --libdir=/usr/x86_64-w64-mingw32/sys-root/mingw/lib
  make install

* cross-compile supagrep
  if you cloned from git repo:
    ./reconfigure.sh --host=x86_64-w64-mingw32

  ./configure --host=x86_64-w64-mingw32
  make


* cross-compile supagrep from cygwin with QT 
	You may have to edit the file to configure the QTDIR environment variable.
	./build_cygwin_mingw_qt.sh


ChangeLog
commit 4324a456c419b57d35e9dff074559cea76f4ff91
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 25 19:30:00 2023 +0100

    Fixed format

commit 286f850decf345ce1a99cdb086eebcc27c2b1f29
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 25 19:07:58 2023 +0100

    Fixed open command

commit bc885ac34ef6307eb1c2b1287bacbfd2f5e9755c
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 25 18:04:53 2023 +0100

    Added editors you may wan't to open with.
    
    not finished yet

commit 5fb6084212d1fb7343e94695e27b1719e9137ac4
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 25 15:41:00 2023 +0100

    Compile forward and addpath only on WIN32 platform

commit c4a313afbbdf570592a88bf396cc9ed531eb7e9c
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Mon Mar 20 21:56:13 2023 +0100

    Fixed FOX build

commit df827c7c394f8f6675c3f8af4e17d55812577aeb
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Mon Mar 20 21:27:10 2023 +0100

    Copy filename to clipboard

commit d3602e4f8941bd5fa94d7f7cb69af0ab90213b27
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Mon Mar 20 19:47:41 2023 +0100

    Open file with default app

commit f6702509b6f61fc5d7607ddb915e65a268306f38
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Mon Mar 20 19:14:46 2023 +0100

    Preparation for context menu

commit 89c18fbf187b1660d40f13125c5bf576a8813ac4
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Mon Mar 20 18:49:29 2023 +0100

    Added addpath and forward to target list

commit 47d65704c36badb96517c6e96da1c7684174ec20
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sun Mar 19 20:51:20 2023 +0100

    switch to shared_ptr for Result::Config

commit 8fdd657781a6c5119d6c9806d15819b0ded4cc56
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sun Mar 19 18:05:31 2023 +0100

    First step to hightlight stuff in result win

commit c696d799e6237231ad2a61f246f17427ee7ea393
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sun Mar 19 16:08:33 2023 +0100

    Display search result in Qt gui

commit 27e4349505d5c52addf21ddde1db0f73c6fb3762
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sun Mar 19 15:53:36 2023 +0100

    Moved appendItem to ResultWinCommon class

commit ac6f1a62c7db0c7000668ac23b8d821ac3e105be
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sun Mar 19 15:30:03 2023 +0100

    Moved stuff into ResultWinCommon class

commit 1684c33f9067c71d01be923f1e1d107bd0c9855b
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 18 22:04:04 2023 +0100

    Added get_whole_line from replace

commit 5e24aaf06a166a5a9a4cd954c43fc604faa43eff
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 18 22:00:26 2023 +0100

    Show some results

commit f67fd615d26547d01e44b52c49586adf0d379c3b
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sat Mar 18 22:00:04 2023 +0100

    Added getline from replace

commit ee869b9d470cc395dd1124c491bfefd5961a1c70
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 15 20:31:03 2023 +0100

    New version (minor changes)

commit 98aee7a8cc808b25b08343eab15b2495ec4645c4
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 15 20:26:56 2023 +0100

    fixed permissions

commit a8a7fd740eb1fdb2c8100fc795593c5585a3c3af
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 15 20:26:45 2023 +0100

    use bugfixed cpputils

commit ca6508df28e14faadfe7ab624d44f857cb2ebe7e
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Sun Mar 12 19:25:38 2023 +0100

    Added ResultWinQt

commit 82637af4beaf6fe719cd713dd8a9847024302f12
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Thu Mar 9 20:47:54 2023 +0100

    Integration of search thread, start and stops a search

commit 08fa46aa59561c7d03cb2a3f059c2203e99e5f1c
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Thu Mar 9 20:46:42 2023 +0100

    Bugfix when setCurrentIndex() is called

commit c80e72c2e9636f2e636cf8a8291e25b9ab4085eb
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Thu Mar 9 08:29:38 2023 +0100

    prefill config from command line

commit c068d1e1734d817ec8f028d7e9c41f2736c8a07d
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Thu Mar 9 08:29:14 2023 +0100

    commented out unused libraries

commit 34e42bc92048f4f8f451753ba81af676f141f2db
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 21:15:08 2023 +0100

    Unified command line parsing
    
    also some WIN32 fixes

commit 8f25dc46f473c2fa42990fea6835f601339bd0b8
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 19:09:11 2023 +0100

    integrated command line parsing to mainqt window

commit 771aa011ac6efbd94d52c08b9fce170520c4c75b
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 18:39:31 2023 +0100

    Added DescComboQt widget for pre defined file search patterns

commit c9f7df7f59476fc91256555ab9d4be104123fee4
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 17:42:44 2023 +0100

    Use bigfixed version (toupper)

commit 375614c5071d2550409ef9beed9c7dba938a80da
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 17:42:25 2023 +0100

    moved Search class to common lib

commit c9f3c8030b04c139d6794d1aec20ed4d77cb72f8
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 15:47:41 2023 +0100

    remove FXString from Search class

commit f615d77660563d1e961dd2be4d784afa490293c3
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 15:27:28 2023 +0100

    switched to std::filesystem::path

commit 5cf9f2c23a3e795b300725a0e3a3cea0ddcf5045
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 10:18:44 2023 +0100

    switched to std::chrono

commit ee77c4ab31c5adb90bbb663a76e5ea9f42b6ec60
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 09:13:45 2023 +0100

    switched to c++ thread

commit 5140d036dcb53418eba6a5846fa9893a73cdb129
Author: Martin Oberzalek <kingleo@gmx.at>
Date:   Wed Mar 8 08:45:26 2023 +0100

    Switched to std::mutex

commit 42027adc2b51a692a52e90fa2b78dc26c497893b
Author: Martin Oberzalek <kingleo@gmx.at>

tools/db.h
/*
 * $Log: db.h,v $
 * Revision 1.3  2006/11/24 09:47:27  wamas
 * -Wshadow Warnings ausgebaut
 *
 * Revision 1.2  2006/03/17 19:53:40  wamas
 * tools_config.h serched in the upper directory
 *
 * Revision 1.1.1.1  2006/03/17 19:49:15  wamas
 * own tools reponsitority
 *
 * Revision 1.2  2006/03/09 00:48:26  wamas
 * Added CVS Log Info
 *
 */

#ifndef TOOLS_DB_H
#define TOOLS_DB_H

#include "../tools_config.h"

#ifdef TOOLS_USE_DB

#include <string>
#include <vector>

namespace Tools {

struct DBRow
{
  std::vector<std::string> names;
  std::vector<std::string> values;
  
  public:
  DBRow( const std::vector<std::string> &names_ )
    : names( names_ )
  {}

  DBRow( const std::vector<std::string> &names_, const std::vector<std::string> &values_ )
    : names( names_ ), values( values_ )
  {}
};

struct DBRowList
{
  std::vector<std::string> names;
  std::vector< std::vector<std::string> > values;
  
  public:
  DBRowList() : names() {}
  DBRowList( const std::vector<std::string> &names_ )
    : names( names_ )
  {}
};

template<typename T> class DBErg
{
 public:
  T row_list;
  bool success;

 public:
  DBErg() : success( false ) {}
  DBErg( const std::vector<std::string> &names )
    : 
    row_list( names ),
    success( true )
      {}      

    bool operator!() const { return !success; }
};

class DB
{
 public:
    DB();
    virtual ~DB();

    virtual const char * error() = 0;
    virtual bool connect( const char *hostname, const char* username, const char* password ) = 0;
    virtual bool select_db( const char *db ) = 0;
    virtual DBErg<DBRowList> query( const std::string &sql ) = 0;
    virtual DBErg<DBRowList> select( const std::string &sql, bool table_names ) = 0;
    virtual void close() = 0;
    virtual int insert_id() = 0;
};

} // namespace Tools

#endif

#endif
